projects/akita-ng-fire/src/lib/collection/collection.service.ts

```diff
@@ -316,7 +322,9 @@ export class CollectionService<S extends EntityState<EntityType, string>, Entity
         // Sync all docs
         const syncs = ids.map(id => {
           const path = `${this.getPath(syncOptions)}/${id}`;
-          return this.db.doc<EntityType>(path).snapshotChanges();
+          return this.db.doc<EntityType>(path).snapshotChanges().pipe(
+            map(value => this.formatFromFirestore(value))
```

Comment:
You don't want to apply the operation on the snapshot, but on the data directly.